collector/slave_status.go

```diff
@@ -61,9 +76,17 @@ func ScrapeSlaveStatus(db *sql.DB, ch chan<- prometheus.Metric) error {
 		for i, col := range slaveCols {
 			if value, ok := parseStatus(*scanArgs[i].(*sql.RawBytes)); ok { // Silently skip unparsable values.
 				ch <- prometheus.MustNewConstMetric(
-					newDesc(slaveStatus, strings.ToLower(col), "Generic metric from SHOW SLAVE STATUS."),
+					prometheus.NewDesc(
+						prometheus.BuildFQName("mysql", slaveStatus, strings.ToLower(col)),
+						"Generic metric from SHOW SLAVE STATUS.",
+						[]string{"master_uuid", "master_host", "channel_name"},
+						nil,
+					),
 					prometheus.UntypedValue,
 					value,
+					columnValue(scanArgs, slaveCols, "Master_UUID"),
```

Comment:
I initially had it outside of for loop, but decided to remove those variables and pay the CPU cost.

Performance wise doing the scan outside of for loop is probably a very minuscule gain. For standard deployment with single replication source it does not make any difference, and for multi-source deployments the cost of going over slaveCols several times is dwarfed by network roundtrip.

But if you prefer to have colIndexes found once, outside of for loop I will change it :)